x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/delayeddatacheck/DatafeedDelayedDataDetector.java

```diff
@@ -134,9 +138,19 @@ private Map<Long, Long> checkCurrentBucketEventCount(long start, long end) {
 
         SearchRequest searchRequest = new SearchRequest(datafeedIndices).source(searchSourceBuilder).indicesOptions(indicesOptions);
         try (ThreadContext.StoredContext ignore = client.threadPool().getThreadContext().stashWithOrigin(ML_ORIGIN)) {
-            SearchResponse response = client.execute(TransportSearchAction.TYPE, searchRequest).actionGet();
+            SearchResponse searchResponse = client.execute(TransportSearchAction.TYPE, searchRequest).actionGet();
             try {
-                List<? extends Histogram.Bucket> buckets = ((Histogram) response.getAggregations().get(DATE_BUCKETS)).getBuckets();
+                Histogram histogram = searchResponse.getAggregations().get(DATE_BUCKETS);
+                if (histogram == null) {
+                    logger.debug(
```

Comment:
We expect this to be a rare occurrence so let's log it at a higher level. Also we want to know if the delayed detector is not working 

```suggestion
                    logger.warn(
```

Reply:
Done.